core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -388,11 +388,9 @@ abstract class RDD[T: ClassTag](
       // Block hit.
       case Left(blockResult) =>
         if (readCachedBlock) {
-          val existingMetrics = context.taskMetrics().inputMetrics
-          existingMetrics.incBytesRead(blockResult.bytes)
```

Comment:
Why? Based on `TaskMetrics.inputMetrics` document, it is

> Metrics related to reading data from a [[org.apache.spark.rdd.HadoopRDD]] or from persisted data, defined only in tasks with input.

Even cached, don't we need to include its size into input size of the task?

Reply:
input size of the task is increased when the rdd is first computed, I think it is not correct to increase the metrics again when read from cache. Correct me if i am wrong.

Reply:
From the doc, I think `inputMetrics` counts all input size the task processes. It is not for input size read from disk. Although you don't need to read from disk if it is cached, it still increases the amount of input the task is going to process.

Reply:
I agree with @viirya, as currently defined, it include all reads. Note, a cached RDD read could still involve network or disk reads.

Reply:
+1 on the @viirya comment.

Reply:
Thanks for the information @viirya. I'll close the pr